appsec.go

```diff
@@ -0,0 +1,259 @@
+package csbouncer
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"gopkg.in/yaml.v2"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	crowdsecAppsecIPHeader   = "X-Crowdsec-Appsec-Ip"
+	crowdsecAppsecURIHeader  = "X-Crowdsec-Appsec-Uri"
+	crowdsecAppsecHostHeader = "X-Crowdsec-Appsec-Host"
+	crowdsecAppsecVerbHeader = "X-Crowdsec-Appsec-Verb"
+	crowdsecAppsecHeader     = "X-Crowdsec-Appsec-Api-Key"
+	crowdsecAppsecUserAgent  = "X-Crowdsec-Appsec-User-Agent"
+)
+
+type Timeout struct {
+	ConnectTimeout        *int `yaml:"connect_timeout"`
```

Comment:
We should probably transform those to `time.Duration` (and let the user of the library handle the parsing/validation). 
With the way things are currently, it's not possible to have a timeout lower than 1s, which is likely way too long for the majority of users.